math/ncr_modulo_p.cpp

```diff
@@ -101,50 +125,71 @@ class NCRModuloP {
             return 1;
         }
         // fac is a global array with fac[r] = (r! % p)
-        int64_t denominator = modInverse(fac[r], p);
-        if (denominator < 0) {  // modular inverse doesn't exist
-            return -1;
-        }
-        denominator = (denominator * modInverse(fac[n - r], p)) % p;
-        if (denominator < 0) {  // modular inverse doesn't exist
+        const auto denominator = (fac[r] * fac[n - r]) % p;
+        const auto denominator_inv = utils::modInverse(denominator, p);
+        if (denominator_inv < 0) {  // modular inverse doesn't exist
             return -1;
         }
-        return (fac[n] * denominator) % p;
+        return (fac[n] * denominator_inv) % p;
     }
 };
 }  // namespace ncr_modulo_p
 }  // namespace math
 
 /**
- * @brief Test implementations
- * @param ncrObj object which contains the precomputed factorial values and
- * ncr function
- * @returns void
+ * @brief tests math::ncr_modulo_p::NCRModuloP
```

Comment:
IIRC, Doxygen won't detect it properly without this, unless there's a documented parameter.
```suggestion
 * @brief tests math::ncr_modulo_p::NCRModuloP
 * @returns void
```

Reply:
This function does not _return void_.

In my opinion the configuration of doxygen should be changed in such a way, that this function would be _detected_. Till then, it is fine, because this function is only running some tests. 

I would rather go for
```
@param void
```
because the declaration of this function could we equivalently written as:
```cpp
static void tests(void)
```
(cf. [godbolt.org](https://godbolt.org/z/699GTrzYP)).

Reply:
I think it was a bug, I'm not 100% sure, though.
@kvedala might know more about it.